packages/manager/apps/pci-instances/src/components/cart/__tests__/Cart.component.spec.tsx

```diff
@@ -0,0 +1,120 @@
+import { render, screen } from '@testing-library/react';
+import { Text } from '@ovhcloud/ods-react';
+import { describe, vi, expect, test } from 'vitest';
+import { Cart, TCartItem } from '../Cart.component';
+
+const instanceDetails = [
+  {
+    name: 'instance localization',
+    description: (
+      <Text preset="heading-6" className="text-[--ods-color-heading]">
+        eu-west-par
+      </Text>
+    ),
+    price: 50,
+  },
+  {
+    name: 'image',
+    description: (
+      <Text preset="heading-6" className="text-[--ods-color-heading]">
+        Linux
+      </Text>
+    ),
+    price: 10,
+  },
+];
+
+const kubeDetails = [
+  {
+    name: 'kube localization',
+    description: (
+      <Text preset="heading-6" className="text-[--ods-color-heading]">
+        DE
+      </Text>
+    ),
+  },
+  {
+    name: 'plan',
+    description: (
+      <Text preset="heading-6" className="text-[--ods-color-heading]">
+        free
+      </Text>
+    ),
+    price: 30,
+  },
+];
+
+const cartItems = [
+  {
+    title: 'instance',
+    name: 'b3_8_28_17_01',
+    details: instanceDetails,
+    expanded: true,
+  },
+  {
+    title: 'kubernetes',
+    name: 'cluster_01_02_03',
+    details: kubeDetails,
+    expanded: false,
+  },
+];
+
+vi.mock('@ovh-ux/manager-react-components', () => ({
+  useCatalogPrice: vi.fn().mockReturnValue({
+    getTextPrice: vi.fn((price: number) => `${price}`),
+    getFormattedHourlyCatalogPrice: vi.fn(),
+    getFormattedMonthlyCatalogPrice: vi.fn(),
+  }),
+}));
+
+const renderCart = (items: TCartItem[]) => {
+  render(<Cart items={items} />);
+  return screen.getByTestId('cart');
+};
```

Comment:
Question (non-blocking): Is it necessary to return a screen.getByTestId ?

It's is used only to expect(cart).toBeInTheDocument(); on every test and it feels useless.
The render method does render the component you pass, so thoses kind of test feels like a library test.

Component testing should test for what's displayed and the behavior when interracting with it.

Reply:
yes, good point thank you!